src/Pos/Block/Error.hs

```diff
@@ -24,11 +26,44 @@ data RollbackException = RollbackTipMismatch HeaderHash HeaderHash
 
 renderRollbackException :: RollbackException -> Text
 renderRollbackException = \case
-    RollbackTipMismatch storedTip attemptedTip->
+    RollbackTipMismatch storedTip attemptedTip ->
         tipMismatchMsg "rollback" storedTip attemptedTip
 
 instance Exception RollbackException where
     displayException = toString . renderRollbackException
 
 instance Buildable RollbackException where
     build = Data.Text.Buildable.build . renderRollbackException
+
+data ApplyBlocksException
+    = ApplyBlocksTipMismatch Text HeaderHash HeaderHash
+    | ApplyBlocksVerifyFailure VerifyBlocksException
+    | ApplyBlocksError Text
+    deriving (Show)
+
+renderApplyBlocksException :: ApplyBlocksException -> Text
+renderApplyBlocksException = \case
+    ApplyBlocksTipMismatch s tip attemptedTip ->
+        tipMismatchMsg s tip attemptedTip
+    ApplyBlocksVerifyFailure e -> renderVerifyBlocksException e
+    ApplyBlocksError e -> e
+
+instance Exception ApplyBlocksException where
+    displayException = toString . renderApplyBlocksException
+
+instance Buildable ApplyBlocksException where
+    build = Data.Text.Buildable.build . renderApplyBlocksException
+
+data VerifyBlocksException
+    = VerifyBlocksError Text
+    deriving (Show)
+
+instance Exception VerifyBlocksException where
+    displayException = toString . renderVerifyBlocksException
+
+instance Buildable VerifyBlocksException where
+    build = Data.Text.Buildable.build . renderVerifyBlocksException
+
+renderVerifyBlocksException :: VerifyBlocksException -> Text
```

Comment:
Why do you define `Buildable` via conversion to `Text` instead of defining conversion to `Text` via `Buildable`? The latter is more reasonable and efficient.